handlers/pulp_rpm/handlers/rpmtools_zypper.py

```diff
@@ -0,0 +1,563 @@
+            log.info('Want to uninstall package %s', package)
+            z.poolRemove(package)
+
+        if z.poolResolve():
+            result = z.tx_summary("uninstall")
+            z.poolCommit(USE_DRY_RUN)
+        z.finish()
+        q.put(result)
+
+    def uninstall(self, names):
+        """
+        Uninstall (erase) packages by name.
+        :param names: A list of package names to be removed.
+        :type names: list
+        :return: Packages uninstalled (erased).
+            {resolved=[Package,],deps=[Package,], failed=[Package,]}
+        :rtype: dict
+        """
+
+        result = None
+
+        # TODO: looks like that this is not necessary for "uninstall". Only str() is required!
+        names = [str(x.replace('-*-*.*', '').replace('*:', '')) for x in names]
+
+        try:
+            log.debug("Starting process for uninstalling %s", names)
+
+            q = Queue()
+            p = Process(target=self.uninstall_p, args=(names, q, ))
+            p.start()
+
+            p.join()
+            result = q.get()
+        except Exception:
+            var = traceback.format_exc()
+            log.error("Exception happend during uninstall: %s", var)
+        finally:
+            log.debug("Uninstall transaction finished.")
+
+        return result
+
+    def update_p(self, names, q):
+        z = Zypp()
+        if names:
+            for package in names:
+                log.info('Want to update package %s', package)
+                z.poolUpdate(package)
+
+                if z.poolResolve():
+                    result = z.tx_summary("update")
+                    z.poolCommit(USE_DRY_RUN)
+        else:
+            log.info('Want to update all packages')
+            z.poolUpdateAll()
+            result = z.tx_summary("update")
+            z.poolCommit(USE_DRY_RUN)
+        z.finish()
+        q.put(result)
+
+    def update(self, names=()):
+        """
+        Update installed packages.
+        When (names) is not specified, all packages are updated.
+        :param names: A list of package names.
+        :type names: [str,]
+        :return: Packages installed (updated).
+            {resolved=[Package,],deps=[Package,], failed=[Package,]}
+        :rtype: dict
+        """
+
+        result = None
+
+        # TODO: looks like that this is not necessary for "update". Only str() is required!
+        names = [str(x.replace('-*-*.*', '').replace('*:', '')) for x in names]
+
+        try:
+            log.debug("Starting process for updating %s", names)
+
+            q = Queue()
+            p = Process(target=self.update_p, args=(names, q, ))
+            p.start()
+
+            p.join()
+            result = q.get()
+        except Exception:
+            var = traceback.format_exc()
+            log.error("Exception happend during update: %s", var)
+        finally:
+            log.debug("Update transaction finished.")
+
+        return result
+
+    def update_minimal(self, advisories=[]):
```

Comment:
Default values should not be mutable. This will need to be changed in other places as well.

http://python-guide-pt-br.readthedocs.io/en/latest/writing/gotchas/#mutable-default-arguments